test/functional/apps/management/_kibana_settings.js

```diff
@@ -30,29 +21,66 @@ export default function ({ getService, getPageObjects }) {
-      });
+    it('should allow setting advanced settings', async function () {
+      await PageObjects.settings.clickKibanaSettings();
+      await PageObjects.settings.setAdvancedSettings('dateFormat:tz', 'America/Phoenix');
+      const advancedSetting = await PageObjects.settings.getAdvancedSettings('dateFormat:tz');
+      expect(advancedSetting).to.be('America/Phoenix');
     });
 
-    after(function () {
-      return PageObjects.settings.clickKibanaSettings()
-      .then(function TestCallSetAdvancedSettingsForTimezone() {
-        screenshots.take('Settings-advanced-tab');
-        log.debug('calling setAdvancedSetting');
-        return PageObjects.settings.setAdvancedSettings('dateFormat:tz', 'UTC');
+    describe('state:storeInSessionStorage', () => {
+      it ('defaults to false', async () => {
+        await PageObjects.settings.clickKibanaSettings();
+        const storeInSessionStorage = await PageObjects.settings.getAdvancedSettings('state:storeInSessionStorage');
+        expect(storeInSessionStorage).to.be('false');
       });
+
+      it('when false, dashboard state is unhashed', async function () {
+        await PageObjects.common.navigateToApp('dashboard');
+        await PageObjects.dashboard.clickNewDashboard();
+        await PageObjects.header.setAbsoluteRange('2015-09-19 06:31:44.000', '2015-09-23 18:31:44.000');
+        const currentUrl = await remote.getCurrentUrl();
+        const urlPieces = currentUrl.match(/(.*)?_g=(.*)&_a=(.*)/);
+        const globalState = urlPieces[2];
+        const appState = urlPieces[3];
+
+        // We don't have to be exact, just need to ensure it's greater than when the hashed variation is being used,
+        // which is around 10 characters.
+        expect(globalState.length).to.be.greaterThan(20);
+        expect(appState.length).to.be.greaterThan(20);
+      });
+
+      it('setting to true change is preserved', async function () {
+        await PageObjects.settings.navigateTo();
+        await PageObjects.settings.clickKibanaSettings();
+        await PageObjects.settings.toggleAdvancedSettingCheckbox('state:storeInSessionStorage');
+        const storeInSessionStorage = await PageObjects.settings.getAdvancedSettings('state:storeInSessionStorage');
+        expect(storeInSessionStorage).to.be('true');
+      });
+
+      it('when true, dashboard state is hashed', async function () {
+        await PageObjects.common.navigateToApp('dashboard');
+        await PageObjects.dashboard.clickNewDashboard();
+        await PageObjects.header.setAbsoluteRange('2015-09-19 06:31:44.000', '2015-09-23 18:31:44.000');
+        const currentUrl = await remote.getCurrentUrl();
+        const urlPieces = currentUrl.match(/(.*)?_g=(.*)&_a=(.*)/);
+        const globalState = urlPieces[2];
+        const appState = urlPieces[3];
+
+        // We don't have to be exact, just need to ensure it's greater than when the hashed variation is being used,
```

Comment:
Ah, copy paste error for this one.  As far as the character count goes - I'm not sure of the stability of the size of the hash so I didn't want to test the exact character count and upped it to 20.  I updated the comment.